src/js/App/GlobalFilter/constants.js

```diff
@@ -84,3 +88,27 @@ export const generateFilter = async () => {
 
     return [data, currToken];
 };
+
+export const flatTags = (filter, encode = false, format = false) => {
+    const { Workloads, SID, ...tags } = filter;
+    const mappedTags = flatMap(
+        Object.entries({ ...tags, ...!format && { Workloads, SID } } || {}),
+        ([namespace, item]) => Object.entries(item || {})
```

Comment:
I think it is more efficient to filter out entries with no values instead of replacing them with empty objects. You could do it inside the filter bellow and remove one iteration.

Reply:
Not really, this one is to check that the item is defined, otherwise it was throwing errors that `item` is not defined. Thiw was caused by default values for `workloads` and `SID`, they can be undefined and so the line 95 will expand to
```JS
Object.entries({
  SID: undefined,
  workloads: undefined
}).map(([ namespace, item ]) => Object.entries(item)).filter() // here `item` can be undefined. 
```

We could pass default values on line 93, but that'll boil down to same thing as in here. Or we could assign `Workloads` or `SID` only when defined, but that feels like more expensive than running `Object.entries` on empty object.

Reply:
Ah got it. :+1: